### PR TITLE
Add rule S3981 ('no-collection-size-mischeck') for TS/JS

### LINKED ISSUE #1512: Add rule S3981 ('no-collection-size-mischeck') for TS/JS
We can make rule rely on types for TS

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #1512

### CODE REVIEW

File: eslint-bridge/src/rules/no-collection-size-mischeck.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+} from "../utils/isRequiredParserServices";
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+
+const CollectionLike = ["Array", "Map", "Set"];
+const CollectionSizeLike = ["length", "size"];
+
+export const rule: Rule.RuleModule = {
+  create(context: Rule.RuleContext) {
+    const services = context.parserServices;
+    if (isRequiredParserServices(services)) {
```
Comment (saberduck) [inline eslint-bridge/src/rules/no-collection-size-mischeck.ts]: we can invert this condition and use early return, so the indentation of the whole function is decreased

File: eslint-bridge/src/rules/no-collection-size-mischeck.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import { Rule } from "eslint";
+import * as estree from "estree";
+import {
+  isRequiredParserServices,
+  RequiredParserServices,
+} from "../utils/isRequiredParserServices";
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+
+const CollectionLike = ["Array", "Map", "Set"];
```
Comment (saberduck) [inline eslint-bridge/src/rules/no-collection-size-mischeck.ts]: can we add also `WeakSet` and `WeakMap` ?

Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/rules/no-collection-size-mischeck.ts]: Yes, we can.

File: eslint-bridge/src/rules/no-collection-size-mischeck.ts (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/rules/no-collection-size-mischeck.ts]: Won't need to do that because the logic changed a bit.

**Pull request merged**